src/site/content/en/blog/fixing-layout-instability/index.md

```diff
@@ -0,0 +1,184 @@
+---
+title: Fixing layout instability
+subhead: A walkthrough of using WebPageTest to identify and fix layout instability issues.
+authors:
+  - rviscomi
+date: 2019-09-26
+hero: layout-comparison.png
+description: |
+  A walkthrough of using WebPageTest to identify and fix layout instability issues.
+tags:
+  - post # post is a required tag for the article to show up in the blog.
+  - webdev
+  - webpagetest
+  - ux
+---
+
+In an earlier post I wrote about [measuring Cumulative Layout Shift](https://dev.to/chromiumdev/measuring-cumulative-layout-shift-cls-in-webpagetest-5cle) (CLS) in WebPageTest. CLS is an aggregation of all layout shifts, so in this post I thought it'd be interesting to dive deeper and inspect each individual layout shift on a page to try to understand what could be causing the instability and actually try to fix the issue(s).
+
+## Measuring layout shifts
+
+Using the Layout Instability API, we can get a list of all layout shift events on a page:
+
+```js
+new Promise(resolve => {
+  new PerformanceObserver(list => {
+    resolve(list.getEntries().filter(entry => !entry.hadRecentInput));
+  }).observe({type: "layout-shift", buffered: true});
+}).then(console.log);
+```
+
+This produces an array of layout shifts that are not preceded by input events:
+
+```json
+[
+  {
+    "name": "",
+    "entryType": "layout-shift",
+    "startTime": 210.78500000294298,
+    "duration": 0,
+    "value": 0.0001045969445437389,
+    "hadRecentInput": false,
+    "lastInputTime": 0
+  }
+]
+```
+
+In this example there was a single very tiny shift of 0.01% at 210ms.
+
+Knowing the time and severity of the shift is useful to help narrow down what could have caused the shift. Let's turn back to [WebPageTest](https://webpagetest.org) for a lab environment to do more testing.
+
+## Measuring layout shifts in WebPageTest
+
+Similar to measuring CLS in WebPageTest, measuring individual layout shifts will require a custom metric. Fortunately, the process is easier now that Chrome 77 is stable. The Layout Instability API is enabled by default, so you should be able to execute that JS snippet on any website within Chrome 77 and get results immediately. In WebPageTest, you can use the default Chrome browser and not have to worry about command line flags or using Canary.
+
+So let's modify that script to produce a custom metric for WebPageTest:
```

Comment:
```suggestion
So let's modify that script to produce a [custom metric for WebPageTest](https://sites.google.com/a/webpagetest.org/docs/using-webpagetest/custom-metrics#TOC-Specifying-custom-metrics-at-test-time) called `[LayoutShifts]`:
```

On second look I think this link will help WPT newbies (like myself) follow along with the unfamiliar code syntax